libraries/c_sdk_v4/integration-test/core_mqtt_system_test.c

```diff
@@ -0,0 +1,1635 @@
+ */
+#define MQTT_PACKET_TYPE_INVALID        ( 0U )
+
+/**
+ * @brief Number of milliseconds in a second.
+ */
+#define MQTT_ONE_SECOND_TO_MS           ( 1000U )
+
+/**
+ * @brief Length of the MQTT network buffer.
+ */
+#define MQTT_TEST_BUFFER_LENGTH         ( 128 )
+
+/**
+ * @brief Sample length of remaining serialized data.
+ */
+#define MQTT_SAMPLE_REMAINING_LENGTH    ( 64 )
+
+/**
+ * @brief Subtract this value from max value of global entry time
+ * for the timer overflow test.
+ */
+#define MQTT_OVERFLOW_OFFSET            ( 3 )
+
+/**
+ * @brief Sample topic filter to subscribe to.
+ */
+#define TEST_MQTT_TOPIC                 "/iot/integration/test"
+
+/**
+ * @brief Sample topic filter 2 to use in tests.
+ */
+#define TEST_MQTT_TOPIC_2               "/iot/integration/test2"
+
+/**
+ * @brief Length of sample topic filter.
+ */
+#define TEST_MQTT_TOPIC_LENGTH          ( sizeof( TEST_MQTT_TOPIC ) - 1 )
+
+/**
+ * @brief Sample topic filter to subscribe to.
+ */
+#define TEST_MQTT_LWT_TOPIC             "/iot/integration/test/lwt"
+
+/**
+ * @brief Length of sample topic filter.
+ */
+#define TEST_MQTT_LWT_TOPIC_LENGTH      ( sizeof( TEST_MQTT_LWT_TOPIC ) - 1 )
+
+/**
+ * @brief Size of the network buffer for MQTT packets.
+ */
+#define NETWORK_BUFFER_SIZE             ( 1024U )
+
+/**
+ * @brief Client identifier for MQTT session in the tests.
+ */
+#ifndef TEST_CLIENT_IDENTIFIER
+    #define TEST_CLIENT_IDENTIFIER    "MQTT-Test"
+#endif
+
+/**
+ * @brief Length of the client identifier.
+ */
+#define TEST_CLIENT_IDENTIFIER_LENGTH        ( sizeof( TEST_CLIENT_IDENTIFIER ) - 1u )
+
+/**
+ * @brief Client identifier for use in LWT tests.
+ */
+#define TEST_CLIENT_IDENTIFIER_LWT           TEST_CLIENT_IDENTIFIER "-LWT"
+
+/**
+ * @brief Length of LWT client identifier.
+ */
+#define TEST_CLIENT_IDENTIFIER_LWT_LENGTH    ( sizeof( TEST_CLIENT_IDENTIFIER_LWT ) - 1u )
+
+/**
+ * @brief Transport timeout in milliseconds for transport send and receive.
+ */
+#define TRANSPORT_SEND_RECV_TIMEOUT_MS       ( 200U )
+
+/**
+ * @brief Timeout for receiving CONNACK packet in milli seconds.
+ */
+#define CONNACK_RECV_TIMEOUT_MS              ( 1000U )
+
+/**
+ * @brief Time interval in seconds at which an MQTT PINGREQ need to be sent to
+ * broker.
+ */
+#define MQTT_KEEP_ALIVE_INTERVAL_SECONDS     ( 5U )
+
+/**
+ * @brief Timeout for MQTT_ProcessLoop() function in milliseconds.
+ * The timeout value is appropriately chosen for receiving an incoming
+ * PUBLISH message and ack responses for QoS 1 and QoS 2 communications
+ * with the broker.
+ */
+#define MQTT_PROCESS_LOOP_TIMEOUT_MS         ( 700U )
+
+/**
+ * @brief The MQTT message published in this example.
+ */
+#define MQTT_EXAMPLE_MESSAGE                 "Hello World!"
+
+/**
+ * @brief Packet Identifier generated when Subscribe request was sent to the broker;
+ * it is used to match received Subscribe ACK to the transmitted subscribe.
+ */
+static uint16_t globalSubscribePacketIdentifier = 0U;
```

Comment:
For the variable naming as well, what is the naming style are you following here? Don't we have to use the FreeRTOS style as integration tests will be owned the Amazon FreeRTOS repo.

Reply:
Let's make a decision on it separately, and if we decide to follow the FreeRTOS style, I can create a separate PR for it.